src/Handler/CurlFactory.php

```diff
@@ -113,6 +113,8 @@ public static function finish(callable $handler, EasyHandle $easy, CurlFactoryIn
         // Return the response if it is present and there is no error.
         $factory->release($easy);
 
+        assert($easy->response !== null);
```

Comment:
The above call to `$factory->release($easy);` may modify `$easy->response`, so the `!$easy->response` check above will not be sufficient here